tools/hive-bigquery/init_script.py

```diff
@@ -0,0 +1,147 @@
+    tracking_database_password = getpass.getpass(
+        'Enter Cloud SQL tracking database password:')
+
+    time_format = datetime.datetime.now().strftime("%Y_%m_%d_%H_%M_%S")
+    hive_bq_comparison_csv = args.hive_table + "_metrics_hive_bq_" + \
+                             time_format + '.csv'
+    hive_bq_comparison_table = args.hive_table + "_metrics_hive_bq_" + \
+                               time_format
+    log_file_name = args.hive_table + "_metrics_hive_bq_" + time_format + '.log'
+
+    logger = logging.getLogger('Hive2BigQuery')
+    logger.setLevel(logging.DEBUG)
+    # Sets log formatter with time,log level,filename,line no and function
+    # name which produced that log statement
+    formatter = logging.Formatter(
+        "[%(asctime)s - %(levelname)s - %(filename)25s:%(lineno)s - %("
+        "funcName)20s() ] %(message)s")
+    log_handler = logging.FileHandler(filename=log_file_name)
+    log_handler.setLevel(logging.DEBUG)
+    log_handler.setFormatter(formatter)
+    # Adds the above defined handler
+    logger.addHandler(log_handler)
+
+    # BigQuery table name defaults to Hive table name if not provided
+    if args.bq_table == '':
+        args.bq_table = args.hive_table
+
+    if args.gcs_bucket_name.startswith('gs://'):
+        args.gcs_bucket_name = args.gcs_bucket_name.split('gs://')[1]
+    if args.gcs_bucket_name[-1] == '/':
+        args.gcs_bucket_name = args.gcs_bucket_name[:-1]
+
+    configuration_properties = {
+        "hive_server_host": args.hive_server_host,
+        "hive_server_port": args.hive_server_port,
+        "hive_server_username": args.hive_server_username,
+        "hive_database": args.hive_database,
+        "hive_table_name": args.hive_table.lower(),
+        "project_id": args.project_id,
+        "dataset_id": args.bq_dataset_id,
+        "bq_table": args.bq_table,
+        "bq_table_write_mode": args.bq_table_write_mode.lower(),
+        "gcs_bucket_name": args.gcs_bucket_name,
+        "incremental_col": args.incremental_col,
+        "use_clustering": args.use_clustering,
+        "tracking_database_host": args.tracking_database_host,
+        "tracking_database_port": args.tracking_database_port,
+        "tracking_database_user": args.tracking_database_user,
+        "tracking_database_db_name": args.tracking_database_db_name,
+        "tracking_database_password": tracking_database_password,
+        "stage_table_name": args.hive_table.lower(),
+        "time_format": time_format,
+        "hive_bq_comparison_csv": hive_bq_comparison_csv,
+        "hive_bq_comparison_table": hive_bq_comparison_table,
+        "log_file_name": log_file_name
+    }
+
+    print "Check the log file {} for detailed logs".format(log_file_name)
+
+    logger.debug("Configuration Properties")
+    logger.debug(configuration_properties)
+
+    # Writes application properties to a file
+    with open('application.properties', 'w') as file_content:
```

Comment:
Why are we doing this? do we ever clean this file up?

Reply:
Changed the input parameters to config file. Now I'm not writing this to the disk.